packages/web/src/components/organisms/UserProfile/index.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useState } from 'react'
+import { Button, Form, Input, Col, Row, Space } from 'antd'
+import { getAccountAddress } from 'src/fixtures/wallet/utility'
+import { useEffectAsync } from 'src/fixtures/utility'
+import { useGetUser, usePostUser } from 'src/fixtures/dev-for-apps/hooks'
+import { sign } from 'src/fixtures/wallet/utility'
+import styled from 'styled-components'
+
+interface Props {}
+
+const Container = styled.div`
+  display: grid;
+  padding: 1rem;
+  max-width: 1120px;
+  margin: auto;
+  grid-gap: 1rem;
+`
+const Section = styled.section`
+  display: grid;
+  padding: 1rem;
+  grid-gap: 0.5rem;
+`
+const Title = styled.div`
+  font-size: 1rem;
+`
+const Text = styled.div`
+  font-size: 2rem;
+`
+const formLayout = {
+  wrapperCol: { span: 22 }
+}
+
+const DisplayName = ({ address, displayName }: { address: string; displayName: string }) => {
+  const { data } = useGetUser(address)
+
+  return (
+    <Section>
+      <Title>Display name</Title>
+      <Text>{displayName || data?.displayName || '(Not set)'}</Text>
+    </Section>
+  )
+}
+
+export const UserProfile = (_: Props) => {
+  const [author, setAuthor] = useState<string>('')
+  const [isComplete, setIsComplete] = useState<boolean>(true)
+  const [displayName, setDisplayName] = useState<string>('')
+  const { postUserHandler: postUser } = usePostUser()
+  const handleSubmitDisplayName = useCallback(
+    async (name: string) => {
+      setIsComplete(false)
+      const message = `submit display name: ${name}`
+      const signature = (await sign(message)) || ''
+      await postUser(name, message, signature, author).then((data: any) => {
+        setDisplayName(data?.displayName)
+      })
+      setIsComplete(true)
+    },
+    [author, postUser]
+  )
+  useEffectAsync(async () => {
+    const accountAddress = (await getAccountAddress()) || ''
+    setAuthor(accountAddress)
+  })
```

Comment:
It would be better to implement this process as custom hooks and manage the state with swr

Reply:
Thanks for feedback comment.

Cloud you aware of the following changes?
https://github.com/dev-protocol/stakes.social/commit/6303fb363bfc95ad20f064b64f88cb3cdd78024e
(Sorry for huge refactoring patch. The changes were so large that we created another temporary branch.)